### DISCUSSION TITLE
enabling dualstack global ipv6 for the pods

### DISCUSSION THREAD

**Discussion by mhazan01** [zifeo/terraform-openstack-rke2#58] [Q&A]
Hello, i've been using your script to successfully deploy a working rke2 cluster on my openstack, now i've been tasked to add ipv6 to the mix for the app we deploy on it. Any idea on how i can adapt the config file to have dualstack ?

**Comment by zifeo** [comment]
@mhazan01 dual stack for incoming or inter-nodes traffic?

**Reply by mhazan01** [reply]
Hello, most important is the incoming traffic, through public ipv6. For inter-nodes, meaning RKE traffic itself ? that's not needed. but i do need the pods to be able to talk ipv6 between themselves too.

**Reply by zifeo** [reply]
It should be supported, happy to accept a contribution in that direction. A few points to have in mind:
- unsure whether you can run incoming ipv6 without having ipv6 within RKE2
- https://docs.rke2.io/networking/basic_network_options#dual-stack-configuration
- https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md#ipv4--ipv6-dual-stack-services
- it is not clear what setup would work great for a load-balancer to support an ipv6 assignment on Infomaniak as there is no floating ip equivalent yet in ipv6
